base/math.jl

```diff
@@ -613,6 +613,7 @@ julia> hypot(3, 4im)
 hypot(x::Number, y::Number) = hypot(promote(x, y)...)
 hypot(x::Complex, y::Complex) = hypot(abs(x), abs(y))
 hypot(x::T, y::T) where {T<:Real} = hypot(float(x), float(y))
+hypot(x::T, y::T) where {T<:Number} = sqrt(abs2(x) + abs2(y))
```

Comment:
Doesn't this overflow/underflow, which `hypot` should _not_ do?

Reply:
At least use something like `abs(x) * sqrt (1 + (y/x)^2)`.